### ISSUE TITLE
Wi-Fi scan throttling: add explanation what the setting does

### ISSUE THREAD

**Issue by Djfe** [mjaakko/NeoStumbler#375]
### Description

Ignore Wi-Fi scan throttling:
I feel like the current setting doesn't tell you what the setting does.
What does it ignore?

I was legit wondering whether Neostumbler checks automatically for scan throttling and scans faster when it can (according to the dev settings). And if I check this setting it will just stop doing that check (ignore what is set in developer settings) and always scan slower.

When the developer setting has been set already to not throttle wifi scans: what is this setting ignoring then? The default Android behavior?

### Suggested implementation

rename the setting
"Scan Wi-Fi faster/more often (drains battery faster but is more accurate)
Requires Wi-Fi scan throttling to be disabled"

above is way too long but contains the information that I feel is missing now (it could be displayed in an overlay that us shown when you click on the setting, like endpoint selection)

### New request

- [X] I have checked that a similar feature hasn't been requested yet

**Comment by skinkie** [comment]
In Android 14 I have not found this setting under the developer options. There is a setting per app for more accurate location scanning using bluetooth and wifi in the background. I therefore wonder if it should be checked for which android versions this is relevant.

**Comment by Djfe** [comment]
Thanks for taking a look, but the description still doesn't explain what "ignore" means 😅

The neostumbler setting can be enabled once throttling is disabled in the developer settings. If the neostumbler setting isn't enabled, the the app won't do additional scans even though it now could.

It's just confusing me because it's a setting on top of a setting and it's inverted compared to the developer setting.

Regardless, it's up to you whether you think, the current description is enough. It's already better than before, so thanks :)

**Comment by mjaakko** [comment]
Yeah, I agree it's slightly confusing still. The main point of that toggle is to prompt the user to go to device settings and first disable scan throttling there. Non-technical users probably won't know about the scan throttling so that's why this is needed

I might have think about a better way to handle scan throttling, especially now that we have scan intervals are adjusted dynamically. Maybe a warning on the main screen and an option in the troubleshooting menu would be better for this. But hopefully the description in the settings makes this more clear for the time being :)

**Comment by skinkie** [comment]
@Djfe the option is not available in my list.

@mjaakko When I have issued `adb shell settings put global wifi_scan_throttle_enabled 0` this is persistent across boots. Throubleshooting also shows wifi scanning available. But the throtling option itself (still) cannot be enabled.

**Comment by skinkie** [comment]
Honor 200 lite. Runs android 14.

**Comment by Djfe** [comment]
In AOSP it's below verbose wifi logging.
I just watched a video that shows how to enable developer settings on your device and the setting wasn't there, like you said.

Just a hunch but I assume Honor removed this from Android because they do their own power saving settings which maybe aren't customizable (or need different adb variables that we don't know).
You are still able to set the variable via adb but I assume the OS doesn't use the variable anymore so it's essentially useless.

Unless you switch your phone or unlock your bootloader and flash a custom rom, you can't scan wifi unthrottled with Neostumbler.
careful: Unlocking your bootloader wipes your phone (factory reset). So a backup is adviced.

Maybe Neostumbler could add an info at some point that tells the user that the option isn't available on MagicOS (Honor phones).

**Comment by skinkie** [comment]
I have also created a bug report with Honor with the request to add this feature.

**Comment by mjaakko** [comment]
@skinkie NeoStumbler uses the standard Android API to check whether Wi-Fi scanning is throttled on Android 11+ https://github.com/mjaakko/NeoStumbler/blob/142f0ba6590c56b634ff249bd5c290e3faeddf51/app/src/main/java/xyz/malkki/neostumbler/extensions/contextExtensions.kt#L47
If this method returns true, NeoStumbler thinks that Wi-Fi scanning is throttled and won't allow to do more scans. As a workaround, we could also check the value of `wifi_scan_throttle_enabled`. But I'm not sure if changing that option through ADB actually does anything if the same option is not present in the settings? Huawei / Honor is notorious for doing these kind of weird optimizations that break normal Android functionality...

**Comment by Djfe** [comment]
is there a way to figure out via neostumbler logs whether throttling is even enabled via the API? Dies it even return true on his device?

**Comment by skinkie** [comment]
What about a build that toggles the functionality regardless of the Context, and the report if there are errors in scanning?

**Comment by skinkie** [comment]
On an older Nokia/HMD device I do have the throttling option, but either by the rendering or by the number of the reports, I have the "feeling" that Towercollector gathers more cell towers. That feeling is based by the number of items on the map in tower collector (unique measurements) versus the total reports in NeoStumbler and the hexagon based rendering. @mjaakko can you comment on this?

**Comment by mjaakko** [comment]
@skinkie The Wi-Fi throttling option only affects Wi-Fi scanning. Android seems to throttle cell towers scans to once per every five seconds – I'm not aware if it's possible to disable this throttling. Doing scans more often will just give cached data from the previous scan

I haven't used TowerCollector nor looked at its source code, so I'm not sure how it does cell tower scanning. To properly analyze whether it actually collects more (usable) data, you would have to run it in parallel with NeoStumbler on the same device and then compare the amount of unique cell IDs in the raw data
